messagebird/http_client.py

```diff
@@ -34,15 +37,16 @@ def request(self, path, method='GET', params=None, format=ResponseFormat.text):
         }
 
         method_switcher = {
-            'DELETE': requests.delete(url, verify=True, headers=headers, data=json.dumps(params)),
-            'GET': requests.get(url, verify=True, headers=headers, params=params),
-            'PATCH': requests.patch(url, verify=True, headers=headers, data=json.dumps(params)),
-            'POST': requests.post(url, verify=True, headers=headers, data=json.dumps(params)),
-            'PUT': requests.put(url, verify=True, headers=headers, data=json.dumps(params))
+            'DELETE': lambda: requests.delete(url, verify=True, headers=headers, data=json_serialize(params)),
+            'GET': lambda: requests.get(url, verify=True, headers=headers, params=params),
+            'PATCH': lambda: requests.patch(url, verify=True, headers=headers, data=json_serialize(params)),
+            'POST': lambda: requests.post(url, verify=True, headers=headers, data=json_serialize(params)),
+            'PUT': lambda: requests.put(url, verify=True, headers=headers, data=json_serialize(params))
         }
         response = method_switcher.get(method, str(method) + ' is not a supported HTTP method')
```

Comment:
Why not simply check if `method in method_switcher`?

Reply:
Please check [gist with the code example](https://gist.github.com/sintah/5b477987b1e0fe714dc6b1799d1e6555)

Without lambda, all the requests are actually called.

Reply:
I misread your comment. I think your way is easier than using lambdas.